src/PropertyMetadata.php

```diff
@@ -53,25 +56,22 @@ public function serialize()
      */
     public function unserialize($str)
     {
-        $this->__unserialize((array) unserialize((string) $str));
+        [$this->class, $this->name] = unserialize($str);
     }
 
     /**
-     * @return array<string>
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingTraversableTypeHintSpecification
      */
     public function __serialize(): array
     {
-        return [
-            $this->class,
-            $this->name,
-        ];
+        return [$this->serialize()];
```

Comment:
this seems to trigger the parent `$this->serialize` method which will then end in a segmentation fault. Instead `$this->serialize` I think we need to go with implementing it seperatly. Still this would crash applicatons not implementing `__serialize` and `__unserialize` method so I would still suggest great a new major release and revert this changes.

Reply:
if we do a major release there is no way to allow applications to support both v2 and v3 versions of this library.

while with this implementation people can just add to their code

```
    public function __serialize(): array
    {
        return [$this->serialize()];
    }

    public function __unserialize(array $data): void
    {
        $this->unserialize($data[0]);
    }
```

and they got it working, or if they do not care about deprecations, it works anyway even without adding this code.